File: lib/src/main/java/io/github/hidekatsu_izuno/pglite_jdbc/pglite/contrib/intarray.java
package io.github.hidekatsu_izuno.pglite_jdbc.pglite.contrib;

import io.github.hidekatsu_izuno.pglite_jdbc.pglite.interface_.Extension;
import io.github.hidekatsu_izuno.pglite_jdbc.pglite.interface_.ExtensionSetupResult;
import java.net.URL;
import java.util.concurrent.CompletableFuture;

public final class intarray {
    public static final Extension<Object> intarray = new Extension<>();

    static {
        intarray.name = "intarray";
        intarray.setup = (_pg, emscriptenOpts, clientOnly) -> {
            var result = new ExtensionSetupResult<Object>();

            result.bundlePath = resource("intarray.tar.gz");
            return CompletableFuture.completedFuture(result);
        };
    }

    private static URL resource(String name) {
        var url = intarray.class.getClassLoader().getResource(name);
        if (url == null) {
            throw new RuntimeException("Extension bundle not found on classpath: " + name);
        }
        return url;
    }

    private intarray() {
    }
}

File: pglite/src/pglite/contrib/intarray.ts
import type {
	Extension,
	ExtensionSetupResult,
	PGliteInterface,
} from "../interface";

const setup = async (_pg: PGliteInterface, _emscriptenOpts: any) => {
	return {
		bundlePath: new URL("intarray.tar.gz", import.meta.url),
	} satisfies ExtensionSetupResult;
};

export const intarray = {
	name: "intarray",
	setup,
} satisfies Extension;
